pkg/updater/service/datadog_agent.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// Package service provides a way to interact with os services
+package service
+
+const (
+	agentUnit         = "datadog-agent.service"
+	traceAgentUnit    = "datadog-agent-trace.service"
+	processAgentUnit  = "datadog-agent-process.service"
+	systemProbeUnit   = "datadog-agent-sysprobe.service"
+	securityAgentUnit = "datadog-agent-security.service"
+	agentExp          = "datadog-agent-exp.service"
+	traceAgentExp     = "datadog-agent-trace-exp.service"
+	processAgentExp   = "datadog-agent-process-exp.service"
+	systemProbeExp    = "datadog-agent-sysprobe-exp.service"
+	securityAgentExp  = "datadog-agent-security-exp.service"
+)
+
+var (
+	stableUnits = []string{
+		agentUnit,
+		traceAgentUnit,
+		processAgentUnit,
+		systemProbeUnit,
+		securityAgentUnit,
+	}
+	experimentalUnits = []string{
+		agentExp,
+		traceAgentExp,
+		processAgentExp,
+		systemProbeExp,
+		securityAgentExp,
+	}
+)
+
+// SetupAgentUnits installs and starts the agent units
+func SetupAgentUnits() error {
```

Comment:
What if this fails in the middle ? Shouldn't we attempt to rollback ?

Reply:
yes, added rollback